README.md

```diff
@@ -1,8 +1,93 @@
 # powdr
-an extended polynomial identity language (PIL) in rust
+an extended polynomial identity language (PIL) and zk-focused assembly in rust
 
 WARNING: This is a proof-of concept implementation. It is missing many internal checks. DO NOT USE FOR PRODUCTION!
 
+## Basic Concept
+
+powdr is a toolkit that helps build zkEVMs and similar proof systems.
+
+It has two main components:
+
+- A polynomial identity language that allows you to define polynomial constraints, lookups, etc.
+- An extensible assembly language to perform dynamic executions.
+  
+Both the frontends and the backends are flexible.
+
+As an example, powdr contains a frontend that enables you to write code in (no-std) rust,
+which gets compiled to RISCV, then to powdr-assembly and finally to PIL.
+
+All stages of this execution are fully automatic, which means you do not need to write
+additional code for witness generation. All witnesses are automatically inferred from the constraints.
+Since the witnesses are inferred, the system can also ensure that there are no additional (unwanted) witnesses.
```

Comment:
`and the system is not underconstrained.`

Reply:
That is the word I was looking for!